artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionFactoryImpl.java

```diff
@@ -764,7 +764,7 @@ private void failoverOrReconnect(final Object connectionID,
                      failoverRetries++;
                      if (failoverRetryPredicate.test(false, failoverRetries)) {
                         waitForRetry(failoverRetryInterval);
-                        failoverRetryInterval = getNextRetryInterval(failoverRetryInterval);
+                        failoverRetryInterval = serverLocator.getNextRetryInterval(failoverRetryInterval, retryIntervalMultiplier, retryInterval);
```

Comment:
I'm not clear on why this passes retryInterval in the function's (maxRetryInterval) argument vs passing the configured maxRetryInterval, is that intentional ?

Reply:
It's a bug. Nice catch!